### PR TITLE
TAS-904/settings

### PR DESCRIPTION
**Pull request by mojgan-a**

### CODE REVIEW

File: src/Nowruz/modules/Settings/components/Account/index.tsx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+                            <SearchDropdown
+                                    id="location"
+                                    value={user?.meta.city}
+                                    isAsync
+                                    loadOptions={user?.meta.onSelectCity}
+                                    defaultOptions
+                                    icon="search-lg"
+                                    hasDropdownIcon={false}
+                                    onChange={(value) => {
+                                    onSelectCity(value);
```
Comment (marjan-ara) [inline src/Nowruz/modules/Settings/components/Account/index.tsx:129]: Can you explain where is 'onSelectCity' defined?

File: src/Nowruz/modules/Settings/components/Account/index.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                <div className={css.borderSection}>
+                    <div className="grid grid-cols-5 gap-4">
+
+                            <label >City</label>
+                        <div className="col-span-2 ...">
+                            <SearchDropdown
+                                    id="location"
+                                    value={user?.meta.city}
+                                    isAsync
+                                    loadOptions={user?.meta.onSelectCity}
```
Comment (marjan-ara) [inline src/Nowruz/modules/Settings/components/Account/index.tsx:124]: onSelectCity is not in meta. Can you explain the logic?

File: src/Nowruz/modules/Settings/components/Password/usePassword.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    const onSubmit: SubmitHandler<Inputs> = async (data) => {        
+        try {
+          await changePassword({ current_password, password })
+          reset();
+          
+        } catch (error) { console.log(error) }
+      };
+
+    // const onSubmit: SubmitHandler<Inputs> = async ({ current_password, password }) => {
```
Comment (marjan-ara) [inline src/Nowruz/modules/Settings/components/Password/usePassword.tsx:51]: remove unnecessary comments

**Review by marjan-ara — CHANGES_REQUESTED** [review]

File: src/Nowruz/modules/settings/components/account/index.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+
+const Account = () => {
+
+    const [modalVisibility, setModalVisibility] = useState(false);
+    let reasonbody = "";
+
+    const onChangeTextHandler = (e:AnyAction) => {
+        reasonbody = e.target.value;
+        console.log('gt',reasonbody);
```
Comment (marjan-ara) [inline src/Nowruz/modules/settings/components/account/index.tsx:21]: Please remove this

File: src/Nowruz/modules/settings/components/account/index.tsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    const closeAccount = () => {
+        deleteAccount(reasonbody);
+    };
+
+    return (
+        <>
+            <div className='hidden'>
+                <div className={css.borderSection}>
+                    <div className="flex flex-row w-full pt-8 items-center">
+                        <h2 className="grow css.title">Account Information</h2>
```
Comment (marjan-ara) [inline src/Nowruz/modules/settings/components/account/index.tsx:33]: I think css.title is not recognized here. Please use this
{ `grow ${css.title} `}

File: src/Nowruz/modules/settings/components/account/index.tsx [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+                    </div>
+                </div>
+                <div className={css.borderSection}>
+                    <div className='grid grid-cols-5 gap-4'>
+                        <label>Username</label>
+                        <div className="col-span-2">
+                            <Input
+                                id="username"
+                                prefix="socious.io/"
+                                onChange={(e) => console.log(e.target.value)}
```
Comment (marjan-ara) [inline src/Nowruz/modules/settings/components/account/index.tsx:72]: please check onChange function, functionality is missed (should check the availability of username), moreover, remove all console.logs before commit

File: src/Nowruz/modules/settings/components/password/index.tsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+                        className="mr-1"
+                        src={isPasswordMatch ? '/icons/green-check.svg' : '/icons/grey-check.svg'}
+                        alt="check"
+                        />
+                        Password match
+                    </div>
+                </div>
+                <div className="grid grid-cols-1 gap-4 place-items-end pt-8">
+                    <div className='flex gap-4'>
+                        <Button color="info" onClick={()=> reset()}>Cancel</Button>
```
Comment (marjan-ara) [inline src/Nowruz/modules/settings/components/password/index.tsx:87]: suggestion: onClick={reset}

File: src/Nowruz/modules/settings/components/account/index.tsx [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+                            multiline
+                            label='Reason (optional)'
+                            customHeight="160px"
+                            onChange={onChangeTextHandler}
+                            placeholder='Please let us know why you are closing your account.'
+                        />
+                    </div>
+                    <div className='flex mt-8 justify-end gap-2'>
+                        <Button onClick={() => { setModalVisibility(false); }} color="info">cancel</Button>
+                        <Button onClick={() => closeAccount()} color="error">
```
Comment (marjan-ara) [inline src/Nowruz/modules/settings/components/account/index.tsx:129]: suggestion: onClick={closeAccount}

File: src/Nowruz/modules/settings/components/password/usePassword.tsx [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+import { useEffect, useState } from 'react';
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { changePassword } from 'src/core/api';
+import { passwordPattern } from 'src/core/regexs';
+import * as yup from 'yup';
+
+import { Inputs } from "./password.type";
+
+const schema = yup.object().shape({
+    current_password: yup.string().required(),
```
Comment (marjan-ara) [inline src/Nowruz/modules/settings/components/password/usePassword.tsx:11]: Add custom error message for validation. e.g. Required for all required fields

File: src/Nowruz/modules/settings/components/password/usePassword.tsx [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+        resolver: yupResolver(schema),
+    });
+    const current_password = watch('current_password');
+    const password = watch('password');
+    const confirmPassword = watch('confirm');
+    useEffect(() => {
+        setIsFormValid(
+            !!password && password.length >= 8 && passwordPattern.test(password) && (password === confirmPassword),
+        );
+        setIsPasswordLengthValid(!!(password && password.length >= 8));
```
Comment (marjan-ara) [inline src/Nowruz/modules/settings/components/password/usePassword.tsx:37]: No need to add these states for validation, Yup has built in functions which can manage validations. for example using regex for password pattern. You can use chatGPT to generate regex and use it in YUP. Also for length and repeatPass you can use YUP

**Review by marjan-ara — CHANGES_REQUESTED** [review]

File: src/Nowruz/modules/settings/components/password/index.tsx (hunk 7 above)
Comment (mojgan-a) [reply src/Nowruz/modules/settings/components/password/index.tsx:87]: i tried your suggestion and it didn't work.

**Review by marjan-ara — APPROVED** [review]

File: src/Nowruz/modules/settings/components/account/index.tsx [hunk 11]
```diff
@@ -0,0 +1,137 @@
+import { AnyAction } from '@reduxjs/toolkit';
+import { useState } from 'react';
+import { Button } from 'src/Nowruz/modules/general/components/Button';
+import { FeaturedIcon } from 'src/Nowruz/modules/general/components/featuredIcon-new';
+import { Input } from 'src/Nowruz/modules/general/components/input/input';
+import { Modal } from 'src/Nowruz/modules/general/components/modal';
+import { SearchDropdown } from 'src/Nowruz/modules/general/components/SearchDropdown';
+import { deleteAccount } from 'src/pages/delete-profile/delete-profile.service';
```
Comment (marjan-ara) [inline src/Nowruz/modules/settings/components/account/index.tsx:8]: deleteAccount is imported from v2 pages, we have removed all v2 pages and here we have a conflict for merging. Please write the logic of each page separately in its own folder, just import APIs from src/core/api

**Review by marjan-ara — CHANGES_REQUESTED** [review]

**Comment by seira-yun** [comment]
@marjan-ara Please remove the services from V2 and use the services from Nowruz and submit for code review by @bastantebom 
Try to merge this today.

**Review by bastantebom — APPROVED** [review]

**Review by marjan-ara — APPROVED** [review]

**Pull request merged**